### PR TITLE
fix: clicking on others should set the reason to "other"

### PR DESCRIPTION
**Pull request by Mohammadshahzaib007**
when reporting an article, after clicking the text field, "Others" should be selected #1371

## Events

No

## Experiment

Did you introduce any new experiments?
No

## Manual Testing
Yes

image

### CODE REVIEW

File: packages/shared/src/components/modals/report/ReportModal.tsx [hunk 1]
```diff
@@ -34,6 +34,12 @@ export function ReportModal({
     onClick: (e) => onReport(e, reason, note),
   };
 
+  const onFocus = useCallback(() => {
+    if (!reason) {
```
Comment (sshanzel) [inline packages/shared/src/components/modals/report/ReportModal.tsx:38]: We need an additional check if the reason is empty (the user hasn't selected anything) as they may just want to add more notes.

**Review by sshanzel — CHANGES_REQUESTED** [review]
Looks amazing! We just need to add one more check before focusing 🥳

File: packages/shared/src/components/modals/report/ReportModal.tsx (hunk 1 above)
Comment (Mohammadshahzaib007) [reply packages/shared/src/components/modals/report/ReportModal.tsx:38]: @sshanzel There is already a check for checking whether the reason is empty,  if it is empty then I am setting it to others

Comment (sshanzel) [reply packages/shared/src/components/modals/report/ReportModal.tsx:38]: Ah, nice. Yeah, I see it now. Didn't get much chance to see which `reason` was this, turns out to be the state that manages it. Well done! 🚀

Comment (sshanzel) [reply packages/shared/src/components/modals/report/ReportModal.tsx:38]: One last thing is, to remove the `useCallback`. We shouldn't need it as far as I see.

**Review by sshanzel — APPROVED** [review]
Looks amazing! I am now approving even though I have left one comment since it is non-blocking. Thank you so much for your contribution 🙏

File: packages/shared/src/components/modals/report/ReportModal.tsx [hunk 2]
```diff
@@ -34,6 +34,12 @@ export function ReportModal({
...
+      setReason(ReportReason.Other);
+    }
+  }, [reason]);
```
Comment (sshanzel) [inline packages/shared/src/components/modals/report/ReportModal.tsx:41]:
```suggestion
  const onFocus = () => {
    if (!reason) {
      setReason(ReportReason.Other);
    }
  };
```

**Pull request merged**